docs/sorting-algorithms/Quick-sort.mdx

```diff
@@ -0,0 +1,252 @@
+---
+title: Quick Sort
+slug: /sorting-algorithms/Quick-sort
+---
+
+# Introduction
+
+Quick Sort is a sorting algorithm which follows Divide and Conquer. In this algorithm it picks an element as a pivot
+and arrange the array according to this pivot element.The condition to arrange the array according to pivot element
+is that all the elements which are smaller than or equal to pivot must lies on the left side of pivot.
+And all the elements which are greater than pivot must lie on the right side of the pivot. 
+This can be done in-place, requiring small additional amounts of memory to perform the sorting.  
+
+# Algorithm
+
+In Quick Sort, we follow the following algorithm:
+
+* first we have to pick a pivot element.
+
+* Then partition the array according to pivot element(smaller or equal elements than pivot must lies on left side and greater on right).
+
+* Then apply same steps 1 & 2 on the left part of pivot and right part of pivot recursively.
+
+* We do step 3 until we left with single element.
+
+# Explanation for Quick Sort
+
+### Example 1
+
+<img src="../../static/img/quick-sort.png" width="500" height="350"/>
+
+```
+
+ Array arr[4]:        
+ | 11 | 22 | 44 | 33 |                               
+ Pivot element=33  
+ After Partition :
+ | 11 | 22  || 33 || 44 |                                             
+ left part= | 11 | 22 | 
+ right part= | 44 |                                      
+ Pivot element= | 22 |
+ left part= | 11 |
+ | 11 || 22 || 33 | 44 |
+  Sorted!! 
+
+```
+### Example 2
+<img src="../../static/img/quick-sort2.png" width="500" height="350"/>
```

Comment:
Remove these images. If you created them, upload on Imgur and cross-reference them here